go/utils.go

```diff
@@ -157,3 +157,25 @@ func WaitForTask(taskRef xenapi.TaskRef, delay int) error {
 	}
 	return nil
 }
+
+func WaitForSRReady(srRef xenapi.SRRef) error {
+	token := ""
+	maxTries := 3
+	eventTypes := []string{"PBD"}
+	for i := 0; i < maxTries; i++ {
```

Comment:
If there are new events, the `event.from` API calls will always return immediately.
I understand the purpose of this is to wait up to timeout e.g. 30 seconds, for the expected state. So the durations of the individual `event.from` calls should be recorded. E.g.
```
for timeout >= 0 {
  start := time.Now()
  event.from(...)
  end := time.Now()
  elapsed := end.Sub(start)
  timeout = timeout - elapsed
}
```

Reply:
After discussion, will add a sleep inside "for".